src/o365/onedrive/commands/report/report-usageaccountdetail.ts

```diff
@@ -0,0 +1,41 @@
+import commands from '../../commands';
+import DateAndPeriodBasedReport from '../../../base/DateAndPeriodBasedReport';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+class OneDriveReportUsageAccountDetailCommand extends DateAndPeriodBasedReport {
+  public get name(): string {
+    return commands.REPORT_USAGEACCOUNTDETAIL;
+  }
+
+  public get usageEndpoint(): string {
+    return 'getOneDriveUsageAccountDetail';
+  }
+
+  public get description(): string {
+    return 'Gets details about OneDrive usage by account';
+  }
+
+  public commandHelp(args: {}, log: (help: string) => void): void {
+    log(vorpal.find(this.name).helpInformation());
+    log(
+      `  Examples:
+      
+    Gets details about OneDrive usage by account for the last week
+      ${commands.REPORT_USAGEACCOUNTDETAIL} --period D7
+
+    Gets details about OneDrive usage by account for May 1, 2019
+      ${commands.REPORT_USAGEACCOUNTDETAIL} --date 2019-05-01
+
+    Gets details about OneDrive usage by account for the last week
+    and exports the report data in the specified path in text format
+      ${commands.REPORT_USAGEACCOUNTDETAIL} --period D7 --output text --outputFile 'C:/report.txt'
+
+    Gets details about OneDrive usage by account for the last week
+    and exports the report data in the specified path in json format
+      ${commands.REPORT_USAGEACCOUNTDETAIL} --period D7 --output json --outputFile 'C:/report.json'
```

Comment:
Should `--outputFile` path not be `C:\report.json`?

Reply:
@garrytrinder 

Yes, indeed. This part has been kept like this since the first introduction of all commands. I would suggest following:
- we perform the changes on each command individually and start now
- we perform a larger change replacing this line for each command once the list of commands has been finalised (we still have around 10 commands under pull)

I would prefer the second one since this might be a little bit more efficient